urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Wouldn't that raise before getting here?  We've parsed the redirect headers; this is just throwing away the response body; but I haven't traced through any of the code yet.

In any case, to be safe, happy to catch and discard the conn, but I'm not completely sure which types of exceptions we should try to catch here.